### PR TITLE
MySQL support

### PR DESCRIPTION
**Pull request by ndenny**
- Support for other SQLalchemy-supported databases
- Added requirements files
- Added support for DB URL with parameters (for unix socket support)
- Added group_by parameter

### CODE REVIEW

File: grice/db_controller.py [hunk 1]
```diff
@@ -158,6 +158,23 @@ def parse_join(join_str, outer_join: bool):
...
 
+def parse_col_names(column_names):
+    """
+    This method takes a string of comma-seperated column names and returns a list of column names.
+
+    :param column_names: string
+    :return: column_names: list
+    """
+    if column_names:
+        columns_dict = OrderedDict()
```
Comment (cache-rules) [inline grice/db_controller.py]: I'm assuming we're using OrderedDict here in order to basically get the functionality of a Set that preserves order, is that correct?

File: grice/db_controller.py [hunk 2]
```diff
@@ -158,6 +158,23 @@ def parse_join(join_str, outer_join: bool):
...
+
+        for column_name in column_names.split(','):
+            columns_dict[column_name.strip()] = True
+
+        column_names = list(columns_dict.keys())
```
Comment (cache-rules) [inline grice/db_controller.py]: This can just be `list(columns_dict)`

File: grice/db_controller.py [hunk 3]
```diff
@@ -212,15 +222,16 @@ def table_api(self, name):
...
 
         try:
             table_info = self.db_service.get_table(name)
         except NotFoundError as e:
             return jsonify(success=False, error=str(e)), 404
 
         try:
-            rows, columns = self.db_service.query_table(name, column_names, page, per_page, filters, sorts, join)
+            rows, columns = self.db_service.query_table(name, column_names, page, per_page, filters, sorts, join, group_by, 
+                format_as_list=request.args.get('_list', '').lower() in ['t', 'true', '1'])
```
Comment (cache-rules) [inline grice/db_controller.py]: Incorrect level of indent on continuation here. Also I would define a variable `format_as_list` above this and pass that in here.

File: grice/db_service.py [hunk 4]
```diff
@@ -325,6 +330,26 @@ def apply_column_sorts(query, table: Table, join_table: Table, sorts: dict):
 
     return query
 
+def apply_column_groups(query, table: Table, join_table: Table, group_by: list):
```
Comment (cache-rules) [inline grice/db_service.py]: I think this should be named `apply_group_by`, thoughts?

File: grice/db_service.py [hunk 5]
```diff
@@ -416,6 +441,8 @@ def query_table(self, table_name, column_names: list=None, page: int=DEFAULT_PAG
         if len(columns) == 0:
             return [], []
 
+        # group_by_cols = names_to_columns(group_by, table, join_table) if group_by else None
```
Comment (cache-rules) [inline grice/db_service.py]: This can be removed.

File: grice/db_service.py [hunk 6]
```diff
@@ -430,16 +457,24 @@ def query_table(self, table_name, column_names: list=None, page: int=DEFAULT_PAG
...
             result = conn.execute(query)
 
             for row in result:
-                data = {}
-
-                for column in columns:
-                    full_column_name = column.table.name + '.' + column.name
-                    column_label = column.table.name + '_' + column.name
-                    data[full_column_name] = row[column_label]
+                if format_as_list:
```
Comment (cache-rules) [inline grice/db_service.py:465]: I like this feature, good idea.

**Review by cache-rules — CHANGES_REQUESTED** [review]
This is great, thanks for these changes. I noted a few minor issues that should be addressed and then I will merge. Let me know if you have any questions (or disagree, open to discussion)

File: grice/db_controller.py (hunk 1 above)
Comment (ndenny) [reply grice/db_controller.py]: I think I copy and pasted your code for parse_sorts for this... :) I was going to ask you why you were doing it this way... but yeah, perhaps just `set(column_names.split(',)` is all that is really required?

File: grice/db_service.py (hunk 4 above)
Comment (ndenny) [reply grice/db_service.py]: Yeah, I'm terrible at naming things.

File: grice/db_controller.py (hunk 3 above)
Comment (ndenny) [reply grice/db_controller.py]: NB: got a change coming where all these parameters are in one object, so we can handle POSTed JSON.

File: grice/db_controller.py (hunk 1 above)
Comment (ndenny) [reply grice/db_controller.py]: Ah - OrderedDict.fromkeys() seems to be the thing I need.

**Pull request merged**